com.reprezen.swagedit/src/com/reprezen/swagedit/json/JsonUtil.java

```diff
@@ -10,37 +10,30 @@
  *******************************************************************************/
 package com.reprezen.swagedit.json;
 
+import com.fasterxml.jackson.core.JsonPointer;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Strings;
 import com.reprezen.swagedit.json.JsonSchemaManager.JSONSchema;
+import com.reprezen.swagedit.json.references.JsonReference;
 
 public class JsonUtil {
 
 	private static final JsonSchemaManager schemaManager = new JsonSchemaManager();
 
 	/**
-	 * Returns true if the node is a reference to another node.
-	 * 
-	 * @param node
-	 * @return true if is reference
-	 */
-	public static boolean isRef(JsonNode node) {
-		return node.isObject() && node.has("$ref");
-	}
-
-	/**
 	 * Returns the node that is referenced by the refNode.
 	 * 
 	 * @param document
 	 * @param refNode
 	 * @return referenced node
 	 */
 	public static SchemaDefinition getReference(JsonNode document, JsonNode refNode) {
-		if (!isRef(refNode) || document == null)
+		if (!JsonReference.isReference(refNode) || document == null) {
 			return new SchemaDefinition(document, refNode);
+		}
 
-		JsonNode found = null;
+		// TODO Make use of JSONReference
```

Comment:
Do we still want to make use of JSONReference? If yes, let's create an issue for this. If no, we can remove the TODO